src/enums/battle-scene-event-type.ts

```diff
@@ -11,6 +27,12 @@ export enum BattleSceneEventType {
    */
   BERRY_USED = "onBerryUsed",
 
+  /**
+   * Triggers when a run end either through winning or losing (but not save and quitting)
+   * @see {@linkcode RunEndEvent}
+   */
+  RUN_END = "onRunEnd",
+
```

Comment:
Not sure if this event name is optimal. It could also be associated with "Running from a battle" by accident.